src/syscheckd/src/create_db.c

```diff
@@ -286,7 +287,12 @@ static void transaction_callback(ReturnTypeCallback resultType, const cJSON* res
     cJSON_AddNumberToObject(data, "version", 2.0);
     cJSON_AddStringToObject(data, "mode", FIM_EVENT_MODE[txn_context->evt_data->mode]);
     cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[txn_context->evt_data->type]);
-    //cJSON_AddNumberToObject(data, "timestamp", time(NULL)); This should be the last_event field
+
+    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL){
+        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
+    }else{
+        cJSON_AddNumberToObject(data, "timestamp", txn_context->latest_entry->file_entry.data->last_event);
+    }
```

Comment:
```suggestion
    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL) {
        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
    } else {
        cJSON_AddNumberToObject(data, "timestamp", txn_context->latest_entry->file_entry.data->last_event);
    }
```

Reply:
Done in [commit](https://github.com/wazuh/wazuh/pull/12359/commits/224cd150ea90378c265d90b500fd7e98d723ab46)